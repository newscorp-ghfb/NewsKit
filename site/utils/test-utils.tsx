/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
