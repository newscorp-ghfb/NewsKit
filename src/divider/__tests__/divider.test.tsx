import {renderToFragmentWithTheme} from '../../test/test-utils';
import {Divider} from '..';

describe('Divider', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(Divider);
    expect(fragment).toMatchSnapshot();
  });
  test('renders as vertical', () => {
    const fragment = renderToFragmentWithTheme(Divider, {
      vertical: true,
    });

    expect(fragment).toMatchSnapshot();
  });
  test(`renders with logical prop overrides`, () => {
    const fragment = renderToFragmentWithTheme(Divider, {
      vertical: true,
      overrides: {
        paddingInline: '30px',
        marginBlock: '20px',
      },
    });
    expect(fragment).toMatchSnapshot();
  });
});
