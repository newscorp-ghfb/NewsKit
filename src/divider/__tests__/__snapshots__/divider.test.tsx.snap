// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders as vertical 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-style: solid;
  border-color: #DEDEDE;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-left-width: 1px;
  display: inline-block;
  height: 100%;
}

<hr
    aria-hidden="true"
    class="emotion-0"
    data-testid="divider"
  />
</DocumentFragment>
`;

exports[`Divider renders with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-style: solid;
  border-color: #DEDEDE;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-top-width: 1px;
  width: 100%;
}

<hr
    aria-hidden="true"
    class="emotion-0"
    data-testid="divider"
  />
</DocumentFragment>
`;

exports[`Divider renders with logical prop overrides 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-style: solid;
  border-color: #DEDEDE;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-left-width: 1px;
  display: inline-block;
  height: 100%;
  margin-block: 20px;
  padding-inline: 30px;
}

<hr
    aria-hidden="true"
    class="emotion-0"
    data-testid="divider"
  />
</DocumentFragment>
`;
